cli/src/parser.sk

```diff
@@ -286,29 +286,43 @@ private fun parse(
         }
       | None() if (!str.startsWith("-") && posArgId < posArgs.size()) ->
         updateValue(values, posArgs[posArgId], str) match {
-        | err @ Some _ ->
-          !error = err;
+        | Failure(err) ->
+          !error = Some(err);
           break void
-        | _ -> void
+        | Success _ -> void
         };
         posArgs[posArgId] match {
         | ArrayArg _ -> void
         | _ -> !posArgId = posArgId + 1
         }
-      | None() -> // Check for short forms that take values such as `-O2`.
+      | None() ->
         !argName = str.take(2);
-        value = str.substring(str.getIter().forward(2));
         options.maybeGet(argName) match {
+        // Check for short forms that take values such as `-O2`.
         | Some(arg) if (arg is ValuedArg _) ->
+          value = str.stripPrefix(argName);
           updateValue(values, arg, value) match {
-          | err @ Some _ ->
-            !error = err;
+          | Failure(err) ->
+            !error = Some(err);
             break void
-          | None() -> void
+          | Success _ -> void
           }
+        // Check for short boolean flags stringed together such as `-abc`.
         | _ ->
-          !error = Some(InvalidArgumentError(str));
-          break void
+          for (flag in str.stripPrefix("-")) {
+            !argName = `-${flag}`;
+            options.maybeGet(argName) match {
+            | Some(arg) if (arg is BoolArg _) ->
+              if (!values.containsKey(arg.name)) {
+                values.set(arg.name, BoolValue(true))
+              } else {
+                !error = Some(DuplicateValueError(arg.name))
+              }
+            | _ ->
+              !error = Some(InvalidArgumentError(str));
+              break void
+            }
+          }
```

Comment:
Alternatively: eat characters as long as they are `BoolArg`s, the possibly remaining one is a `ValuedArg` in short form. E.g. `-abcO2`

Reply:
Alright, I'll update!